pkgs/by-name/ei/eiffelstudio/package.nix

```diff
@@ -0,0 +1,125 @@
+{
+  copyDesktopItems,
+  fetchurl,
+  lib,
+  makeDesktopItem,
+  makeWrapper,
+  pkgs,
+  stdenv,
+
+  dotnetSupport ? false, # Untested integration with .NET, disabled for now
+}:
+
+let
+  platformName =
+    {
+      "x86_64-linux" = "linux-x86-64";
+      "aarch64-linux" = "linux-arm64";
+      "x86_64-darwin" = "macosx-x86-64"; # Doesn't compile with clang yet
+      "aarch64-darwin" = "macosx-armv6"; # Need to do more testing on macOS
+    }
+    .${stdenv.system} or null;
+in
+stdenv.mkDerivation rec {
+  pname = "eiffelstudio";
+  version = "25.02.98732";
+
+  src = fetchurl {
+    url = "https://ftp.eiffel.com/pub/download/latest/pp/PorterPackage_${lib.versions.majorMinor version}_rev_${lib.versions.patch version}.tar";
+    sha256 = "sha256-/XoewqCeh1NfB3ve9UL+0WZfZ5DEa4N7REl67Ftlxt0=";
+  };
+
+  nativeBuildInputs =
+    with pkgs;
+    [
+      coreutils
+      copyDesktopItems
+      gnused
+      makeWrapper
+      pax
+      pkg-config
+    ]
+    ++ lib.optional dotnetSupport dotnet-sdk;
+
+  buildInputs =
+    with pkgs;
+    [
+      adwaita-icon-theme
+      cairo
+      gdk-pixbuf
+      glib
+      gtk3
+      pango
+      pkg-config
```

Comment:
pkg-config is only needed in nativeBuildInputs

Reply:
That makes sense, so I've removed it. However, I do have one small question. This application also makes use of a compiler, and maybe pkg-config, gtk3 at runtime. It depends on whether you're building a console or graphical application. Basically it's the main purpose of this software. How would you recommend I run it then, through my own shell or flake that brings in these dependencies around the app?

Or should I be using wrapGAppsHook or similar? Sorry, I am completely new to the codebase. I tried to look at other IDEs to see how this might be handled, but I'm not sure.

In general, the application expects headers and libraries from GTK/X11 to be available, and it relies on pkg-config to find them IIRC.

Reply:
Ok, it turned into not such a small question :)